pkg/controller/node/node_controller_test.go

```diff
@@ -429,129 +429,63 @@ func TestGetCandidateMachines(t *testing.T) {
 		},
 		expected: nil,
 	}, {
-		//progress all available nodes
+		// node-2 is going to change config, so we can only progress one more
 		progress: 2,
 		nodes: []*corev1.Node{
 			newNodeWithReady("node-0", "v1", "v1", corev1.ConditionTrue),
-			newNodeWithReady("node-1", "v1", "v1", corev1.ConditionFalse),
+			newNodeWithReady("node-1", "v1", "v1", corev1.ConditionTrue),
 			newNodeWithReady("node-2", "v0", "v1", corev1.ConditionTrue),
 			newNodeWithReady("node-3", "v0", "v0", corev1.ConditionTrue),
 			newNodeWithReady("node-4", "v0", "v0", corev1.ConditionTrue),
 		},
-		expected: []*corev1.Node{newNodeWithReady("node-3", "v0", "v0", corev1.ConditionTrue), newNodeWithReady("node-4", "v0", "v0", corev1.ConditionTrue)},
+		expected: []string{"node-3"},
 	}, {
-		//progress nodes but limited
+		// We have a node working, don't start anything else
 		progress: 1,
 		nodes: []*corev1.Node{
 			newNodeWithReady("node-0", "v1", "v1", corev1.ConditionTrue),
-			newNodeWithReady("node-1", "v1", "v1", corev1.ConditionFalse),
+			newNodeWithReady("node-1", "v1", "v1", corev1.ConditionTrue),
 			newNodeWithReady("node-2", "v0", "v1", corev1.ConditionTrue),
 			newNodeWithReady("node-3", "v0", "v0", corev1.ConditionTrue),
 			newNodeWithReady("node-4", "v0", "v0", corev1.ConditionTrue),
 		},
-		expected: []*corev1.Node{newNodeWithReady("node-3", "v0", "v0", corev1.ConditionTrue)},
+		expected: nil,
```

Comment:
Notice how this case changed - before even though we had one node already targeting `v1` we'd go ahead and change desired for `node-3` too.